test/js/upload-wizard/validations.test.js

```diff
@@ -0,0 +1,212 @@
+      upload_file_name: 'bar.txt',
+      _id: '11'
+    }, file]
+    const msgs = validateFile({ file, allFiles, allowedFileTypes: ['.txt'] })
+    expect(msgs.fileName).toEqual('A file named bar.txt already exists in your study')
+  })
+})
+
+describe('upload file validation new file checks', () => {
+  it('requires a file selected to be uploaded', async () => {
+    const file = {
+      status: 'new',
+      name: 'unique',
+      _id: '10'
+    }
+    const allFiles = [{
+      status: 'uploaded',
+      name: 'special',
+      upload_file_name: 'bar.txt',
+      _id: '11'
+    }, file]
+    const msgs = validateFile({ file, allFiles, allowedFileTypes: ['.txt'] })
+    expect(msgs.uploadSelection).toEqual('You must select a file to upload')
+  })
+
+  it('checks required file extensions', async () => {
+    const file = {
+      status: 'new',
+      name: 'unique',
+      uploadSelection: { name: 'bar.txt33' },
+      _id: '10'
+    }
+    const allFiles = [{ file }]
+    const msgs = validateFile({ file, allFiles, allowedFileTypes: ['.txt'] })
+    expect(msgs.fileName).toEqual('Allowed extensions are .txt')
+  })
+
+  it('checks required file extensions are at the end of the file', async () => {
+    const file = {
+      status: 'new',
+      name: 'unique',
+      uploadSelection: { name: 'f.txt.bar' },
+      _id: '10'
+    }
+    const allFiles = [{ file }]
+    const msgs = validateFile({ file, allFiles, allowedFileTypes: ['.txt', '.txt.gz'] })
+    expect(msgs.fileName).toEqual('Allowed extensions are .txt .txt.gz')
+  })
+
+  it('requires a bundle parent to be saved first', async () => {
+    const file = {
+      status: 'new',
+      name: 'cluster labels',
+      options: {
+        cluster_file_id: 'newFile-1'
+      },
+      _id: '10'
+    }
+    const allFiles = [{
+      status: 'new',
+      name: 'Cluster A',
+      _id: 'newFile-1'
+    }, file]
+    const msgs = validateFile({ file, allFiles, allowedFileTypes: ['.txt'] })
+    expect(msgs.parentSaved).toEqual('Parent file must be saved first')
+  })
+})
+
+describe('it checks presence of required fields', () => {
+  it('passes if required fields are present', async () => {
+    const file = {
+      name: 'unique',
+      taxon_id: '1345',
+      _id: '10'
+    }
+    const allFiles = [{ file }]
+    const msgs = validateFile({
+      file, allFiles, allowedFileTypes: ['.txt', '.txt.gz'],
+      requiredFields: [{ label: 'species', propertyName: 'taxon_id' }]
+    })
+    expect(msgs.taxon_id).toEqual(undefined)
+  })
+
+  it('fails if required fields are not present', async () => {
+    const file = {
+      status: 'new',
+      _id: '10'
+    }
+    const allFiles = [{ file }]
+    const msgs = validateFile({
+      file, allFiles, allowedFileTypes: ['.txt', '.txt.gz'],
+      requiredFields: [{ label: 'species', propertyName: 'taxon_id' }]
+    })
+    expect(msgs.taxon_id).toEqual('You must specify species')
+    expect(msgs.name).toEqual('You must specify name')
```

Comment:
I'm getting a JS test failure at this point, but for some reason builds aren't running in Travis for this PR.  Not sure what the issue is.
```
Summary of all failing tests
 FAIL  test/js/upload-wizard/validations.test.js (12.178 s)
  ● it checks presence of required fields › fails if required fields are not present

    expect(received).toEqual(expected) // deep equality

    Expected: "You must specify name"
    Received: undefined

      172 |     })
      173 |     expect(msgs.taxon_id).toEqual('You must specify species')
    > 174 |     expect(msgs.name).toEqual('You must specify name')
          |                       ^
      175 |   })
      176 |
      177 |   it('messages both if multiple required fields are not present', async () => {

      at Object.<anonymous> (test/js/upload-wizard/validations.test.js:174:23)
```